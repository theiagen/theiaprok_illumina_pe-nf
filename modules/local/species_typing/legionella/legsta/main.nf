process LEGSTA {
    tag "$meta.id"
    label 'process_low'

    container "us-docker.pkg.dev/general-theiagen/biocontainers/legsta:0.5.1--hdfd78af_2"

    input:
    tuple val(meta), path(assembly)

    output:
    tuple val(meta), path("*.tsv"), emit: legsta_results
    tuple val(meta), path("LEGSTA_SBT.txt"), emit: legsta_predicted_sbt
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    legsta \\
      ${assembly} > ${prefix}.tsv
    
    # parse outputs
    SBT=""
    if [ ! -f ${prefix}.tsv ]; then
      SBT="No SBT predicted"
    else
      SBT="ST\$(tail -n 1 ${prefix}.tsv | cut -f 2)"
        if [ "\$SBT" == "ST-" ]; then
          SBT="No SBT predicted"
        else
          if [ "\$SBT" == "ST" ]; then
            SBT="No SBT predicted"
          fi
        fi  
    fi

    echo \$SBT > LEGSTA_SBT.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        legsta: \$(legsta --version 2>&1 | sed 's/^.*legsta //; s/ .*\$//;')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch "${prefix}.tsv"
    touch "LEGSTA_SBT.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        legsta: \$(\$(legsta --version 2>&1) | sed 's/^.*legsta //; s/ .*\$//;')
    END_VERSIONS
    """
}