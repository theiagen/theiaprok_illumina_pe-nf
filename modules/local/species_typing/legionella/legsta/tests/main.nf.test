nextflow_process {
    name "Test Process LEGSTA"
    script "../main.nf"
    process "LEGSTA"

    
    test("legsta test") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file("$projectDir/bin/2022AZMC-0005_filtered_contigs.fasta", checkIfExists: true) // Point to our own test data once available in our datasets repo
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.legsta_results,
                    process.out.legsta_predicted_sbt,
                    process.out.versions
                    ).match() }
            )
        }
    }
    
    test("spatyper - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file("$projectDir/bin/2022AZMC-0005_filtered_contigs.fasta", checkIfExists: true) // Point to our own test data once available in our datasets repo
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}