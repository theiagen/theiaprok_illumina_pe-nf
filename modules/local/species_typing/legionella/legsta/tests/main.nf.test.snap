{
    "legsta test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.tsv:md5,fd11c1d8fb9ec4661c73bc8b8ce36938"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "LEGSTA_SBT.txt:md5,c63eb523155fa6df694f6f079046d83f"
                ]
            ],
            [
                "versions.yml:md5,a853be87d21977365afe63ef07450a44"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.8"
        },
        "timestamp": "2025-06-25T21:02:37.0448495"
    },
    "spatyper - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "LEGSTA_SBT.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,5d30d7d639efc717a40a637b739447b9"
                ],
                "legsta_predicted_sbt": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "LEGSTA_SBT.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "legsta_results": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5d30d7d639efc717a40a637b739447b9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.8"
        },
        "timestamp": "2025-06-25T21:02:41.597275987"
    }
}